app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/DetailActivity.java

```diff
@@ -2,83 +2,41 @@
-import android.support.v7.widget.LinearLayoutManager;
-import android.support.v7.widget.RecyclerView;
-import android.support.v7.widget.Toolbar;
-import android.view.View;
 
-import com.example.nvdovin.weatherapp.R;
-import com.example.nvdovin.weatherapp.domain.service.CityService;
-import com.example.nvdovin.weatherapp.domain.service.WeatherDataService;
 import com.example.nvdovin.weatherapp.presentation.application.WeatherApplication;
-import com.example.nvdovin.weatherapp.presentation.details.adapter.MainRecyclerAdapter;
-import com.example.nvdovin.weatherapp.presentation.history.HistoryActivity;
+import com.example.nvdovin.weatherapp.presentation.details.core.DetailView;
+import com.example.nvdovin.weatherapp.presentation.details.core.DetailsPresenter;
+import com.example.nvdovin.weatherapp.presentation.details.dagger.DaggerDetailComponent;
+import com.example.nvdovin.weatherapp.presentation.details.dagger.DetailModule;
 
 import javax.inject.Inject;
 
-import butterknife.BindView;
-import butterknife.ButterKnife;
-
 public class DetailActivity extends AppCompatActivity {
     private static final String DETAIL_BUNDLE = "DETAIL_BUNDLE";
     private static final String TIMESTAMP_KEY = "TIMESTAMP_KEY";
-    private static final String ARGS_KEY = "HISTORY_ARGS";
     private static final String CITY_ID_KEY = "CITY_ID_KEY";
-    private static final int NUMBER_OF_DAYS_TO_FORECAST = 4;
-
-    @BindView(R.id.detail_main_recycler_view)
-    RecyclerView mainRecycler;
-    @BindView(R.id.detail_main_toolbar)
-    Toolbar toolbar;
-    @BindView(R.id.detail_history_button)
-    FloatingActionButton floatingActionButton;
-    @BindView(R.id.detail_main_collapsing_toolbar)
-    CollapsingToolbarLayout collapsingToolbarLayout;
-
 
-    CityService cityService;
+    @Inject
+    DetailsPresenter detailsPresenter;
+    @Inject
+    DetailView detailView;
 
-    WeatherDataService weatherDataService;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.detail_main_layout);
-
-        ButterKnife.bind(this);
 
         Intent intent = getIntent();
         Bundle bundle = intent.getBundleExtra(DETAIL_BUNDLE);
-        final Long cityId = bundle.getLong(CITY_ID_KEY);
-        Long timestamp = bundle.getLong(TIMESTAMP_KEY);
-
-        DetailsPresenter detailsPresenter = new DetailsPresenter(WeatherApplication.getAppComponent().cityService(), WeatherApplication.getAppComponent().weatherDataService());
 
-        collapsingToolbarLayout.setTitle(detailsPresenter.getCityName(cityId));
-        collapsingToolbarLayout.setCollapsedTitleTextColor(Color.WHITE);
-        collapsingToolbarLayout.setExpandedTitleColor(Color.WHITE);
+        DaggerDetailComponent.builder()
+                .appComponent(WeatherApplication.getAppComponent())
+                .detailModule(new DetailModule(this, bundle.getLong(CITY_ID_KEY), bundle.getLong(TIMESTAMP_KEY)))
```

Comment:
Take a look at how I start activities. Have a static "start" method that takes anything you need and then add to the bundle. The module only takes the activity and that's it.